### ISSUE TITLE
jump to next file after deleting a file

### ISSUE THREAD

**Issue by michael-mhrdd** [QL-Win/QuickLook#850]
**Is your feature request related to a problem? Please describe.**
When I delete a file that's open in quicklook, quicklook keeps the displaying the file. For quicklook to display the next file in the list, I either have to mouse-select it, or use down and up keys to re-select the file in explorer.

**Describe the solution you'd like**
After deletion of a file currently previewed  in quicklook, I would like quicklook to automatically update to display the next file in the explorer file list.

**Describe alternatives you've considered**
Can't think of anything.

**Additional context**
This may be a limitation of Windows explorer, which does not fully select the next file after deletion of the previous file in the list.

**Comment by davidcm** [comment]
I'm a new user and this is the only thing I'm missing for now. Sadly this issue is quite old already, I hope my comment brings it into the devs attention.

Instead of previewing the next file, let me add an alternative behaviour: what Quick Look in macOS does is switch the preview to its parent folder because the window changes to have no selection. That's consistent with what happens when the user simply clicks in an empty space to deselect a file/folder icon (which in this Quick Look project doesn't happen either, and should for consistence).

In other words, Apple's Quick Look considers the active item to be:
- if active window has a selection, take the selection for preview
- if active window doesn't have a selection, take the window's represented folder for preview

Just think of an Explorer window to be an open folder, and clicking its empty space is clicking the folder itself.

Hope it makes sense :)

**Comment by Waleedov** [comment]
This is a must-feature!

**Comment by xupefei** [comment]
Previewing the next file is not the right action, because QuickLook should always preview the file that is currently selected, and when such a file has been deleted, Explorer will not switch the selection to any file. In such a case QuickLook should not preview "the next file" either simply because that file is not selected.


> Instead of previewing the next file, let me add an alternative behaviour: what Quick Look in macOS does is switch the preview to its parent folder because the window changes to have no selection. That's consistent with what happens when the user simply clicks in an empty space to deselect a file/folder icon (which in this Quick Look project doesn't happen either, and should for consistence).

This alternative makes sense. The implementation will be like this:

1. Check if the current foreground window belongs to Explorer.
    1. If yes, try to get the path of the file that is being selected.
    2. If no, try to get the path of the folder that is being displayed.
2. Show a preview.

Such a logic is now handled by `QuickLook.Native`, and only (1.ii) is missing right now.
Added to my backlog.

**Comment by Waleedov** [comment]
This is how photo view programs handle it, in this video Windows Photos app moves to the next file after you delete an image. ACDSee app does the same thing and many more.
From a UX perspective, it's more convenient and useful to jump to auto-advance to the next file, we can easily clean up files and folders this way.

https://github.com/QL-Win/QuickLook/assets/48418015/475d05bb-a709-4869-b231-56135676a493

**Comment by emako** [comment]
There are currently no plans to implement this unless xupefei do it
